core/routes/announcements.js

```diff
@@ -0,0 +1,93 @@
+/* @file announcements.js
+ * @brief Defines the handlers for the announcement API endpoints.
+ *
+ * @author Oscar Bezi (bezi@scottylabs)
+ */
+'use strict';
+
+var db;
+
+//==============================================================================
+// Express handlers
+//==============================================================================
+var handlers = {};
+
+//==============================================================================
+// GET/POST/DELETE /announcements
+//==============================================================================
+
+/* @brief Return every announcement. */
+handlers.get = function (req, res) {
+  var query = 'SELECT * FROM announcements';
+  db.query(query, [])
+  .then((rows) => {
+    var data = {};
+    rows = rows.map((row) => {
+      row.id = row.announcement_id;
+      delete row.announcement_id;
+
+      data[row.id] = row;
+      return row;
+    });
+
+    res.status(200);
+    res.json(rows);
+  }).catch((err) => {
+    res.status(500);
+    res.end(`${ err }\n`);
```

Comment:
Was this intentional? This coerces the `err` value to a string. The alternative would be to just call `res.end(err)` and let Express respond with JSON.